homeassistant/components/idasen_desk/button.py

```diff
@@ -0,0 +1,74 @@
+"""Representation of Idasen Desk buttons."""
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DeskData, IdasenDeskCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Idasen Desk connection buttons from config entry."""
+    data: DeskData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            IdasenDeskConnectButton(data.address, data.device_info, data.coordinator),
+            IdasenDeskDisconnectButton(
+                data.address, data.device_info, data.coordinator
+            ),
+        ]
+    )
+
+
+class IdasenDeskConnectButton(ButtonEntity):
+    """Representation of a connect button entity."""
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_icon = "mdi:bluetooth-connect"
+
+    def __init__(
+        self,
+        address: str,
+        device_info: DeviceInfo,
+        coordinator: IdasenDeskCoordinator,
+    ) -> None:
+        """Initialize the connect button."""
+        self._attr_name = f"{device_info[ATTR_NAME]} Connect"
+        self._attr_unique_id = f"connect-{address}"
+        self._attr_device_info = device_info
```

Comment:
It looks like the pattern is going to be common so it might be nicer to have a base entity class in an entity.py that took care of setting the properties that are the same

Reply:
With the ButtonEntityDescription change, this is not duplicated anymore, so I think it is fine to leave it in button.py for now?

Reply:
Its fine for this PR, next PR should be to create an `entity.py` for a shared base class ie `IdenDeskEntity` before adding any more platforms

Reply:
@bdraco  I was now looking into this for another PR, but I am not sure what the IdasenDeskEntity class would have. Only a single line for the `self._attr_device_info = device_info` ? Because everything else is not common for the current platforms.

Reply:
If there isn't anything common yet then it might not worth be doing until there is.  Usually at some point there is